### PR TITLE
Game Success Screen

### PR DESCRIPTION
**Pull request by mlytvynchuk**
Styled success screen

### CODE REVIEW

File: app/src/main/java/com/google/droidjump/DroidStartView.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+public class DroidStartView extends View {
+
+    public DroidStartView(Context context) {
+        super(context);
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
+        super.onDraw(canvas);
+        Bitmap fullDroidPic = decodeResource(getResources(), R.mipmap.droid);
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/DroidStartView.java:35]: nit: fullDroidPicture

File: app/src/main/java/com/google/droidjump/DroidStartView.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        int droidCountOnFullDroidPic = 9;
+        int droidStep = (int) fullDroidPic.getWidth() / droidCountOnFullDroidPic;
+        int droidCharacterStandingNumber = 4;
+        Bitmap droid = createBitmap(fullDroidPic, /* 4th droid on full droid picture */droidStep * droidCharacterStandingNumber, 0, droidStep,
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/DroidStartView.java:39]: /* 4th droid on full droid picture */droidStep * droidCharacterStandingNumber, -- I would extract this into a variable and add a comment explaining what it does.

File: app/src/main/java/com/google/droidjump/DroidStartView.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                fullDroidPic.getHeight());
+        canvas.drawBitmap(droid, 0, getHeight() - droid.getHeight(), /* paint= */null);
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/DroidStartView.java:41]: getHeight() - droid.getHeight() -- same as above

File: app/src/main/java/com/google/droidjump/GameFailureFragment.java [hunk 4]
```diff
@@ -30,7 +30,7 @@
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        View rootView = inflater.inflate(R.layout.game_failure_screen, container, false);
+        View rootView = inflater.inflate(R.layout.game_failure_screen, container, /* attachToRoot= */false);
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameFailureFragment.java:33]: add space between comment and false; same throughout the file.

File: app/src/main/res/layout/how_to_play_screen.xml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <TextView
+            android:id="@+id/how_to_play_header"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginLeft="@dimen/fab_margin"
+            android:layout_marginStart="@dimen/fab_margin"
+            android:layout_marginTop="@dimen/fab_margin"
```
Comment (redls) [inline app/src/main/res/layout/how_to_play_screen.xml:35]: maybe create a new style?

File: app/src/main/java/com/google/droidjump/DroidStartView.java (hunk 2 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/DroidStartView.java:39]: Makes sense

File: app/src/main/res/layout/how_to_play_screen.xml (hunk 5 above)
Comment (mlytvynchuk) [reply app/src/main/res/layout/how_to_play_screen.xml]: Good idea, fixed.

File: app/src/main/java/com/google/droidjump/DrawableElement.java [hunk 6]
```diff
@@ -18,6 +18,11 @@
 
 import android.graphics.Bitmap;
 
+/**
+ * Represents a parent class of all game elements.
+ *
+ * @author maksme@google.com
```
Comment (saimonsaret) [inline app/src/main/java/com/google/droidjump/DrawableElement.java:24]: Laura, do we actually need authors?

File: app/src/main/java/com/google/droidjump/Droid.java [hunk 7]
```diff
@@ -20,34 +20,43 @@
...
 import android.graphics.BitmapFactory;
 
+/**
+ * Represents Droid character.
+ *
+ * @author maksme@google.com
+ */
 public class Droid extends DrawableElement {
 
+    // the variable is responsible for how much time ticks full animation of droid goes
```
Comment (saimonsaret) [inline app/src/main/java/com/google/droidjump/Droid.java:30]: It seems like you have merged incorrectly, there are a lot of unnecessary changes. Please fix them first.

File: app/src/main/java/com/google/droidjump/DrawableElement.java (hunk 6 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/DrawableElement.java:24]: Can delete it, just googled how to make java document comments and there were specified author in examples.

File: app/src/main/java/com/google/droidjump/Droid.java [hunk 8]
```diff
@@ -20,34 +20,40 @@
...
     private int initialY;
     private int jumpHeight;
-    // so that droid can easily jump over all obstacles we need to add two measures: the highest obstacle + additional height
-    private final static int additionalHeight = 50;
     // the variable is responsible for how much time ticks full animation of droid goes
     public final static int fullAnimationTicks = 4;
     // the variable is responsible for how much time ticks must pass to change a bitmap so that animate droid smoothly
     public final static int animationStepTicks = 2;
-
+    // so that droid can easily jump over all obstacles we need to add two measures: the highest obstacle + additional height
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/Droid.java:37]: nit: Let's have the comments start with capital letter. eg. // So that ..

File: app/src/main/java/com/google/droidjump/Droid.java [hunk 9]
```diff
@@ -20,34 +20,40 @@
...
+    private final static int additionalHeight = 50;
 
     public Droid(int x, int y, Resources resources) {
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/Droid.java:40]: I am confused, why don't we use this why for this.y; where is the assignment?

File: app/src/main/java/com/google/droidjump/GameConstants.java [hunk 10]
```diff
@@ -15,11 +15,16 @@
...
+/**
+ * Stores all global constants of the game
+ */
+public interface GameConstants {
     int DROID_COUNT_ON_FULL_DROID_PICTURE = 9; // Count of droids at res/mipmap/droid.png image
     int DROID_JUMPING_CHARACTER_INDEX = 4; // Index of droid at res/mipmap/droid.png image
     int DROID_FIRST_STEP_INDEX = 5; // Index of droid at res/mipmap/droid.png image
     int DROID_SECOND_STEP_INDEX = 6; // Index of droid at res/mipmap/droid.png image
     int SLEEP_TIME = 15; // Sleep time in gameView (ms)
+    // String for identifying current level from SharedPreferences
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameConstants.java:28]: add a new line.

File: app/src/main/java/com/google/droidjump/GameFragment.java [hunk 11]
```diff
@@ -33,9 +39,19 @@
...
+
+        // Writing size to the screen variable
+        defaultDisplay.getSize(screen);
+
+        // Extracting screen coordinates
+        int screenX = screen.x;
+        int screenY = screen.y;
+
+        // Initializing gameView
+        gameView = new GameView(getActivity(), screenX, screenY, /* isPlaying= */ true);
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameFragment.java:54]: nit: use activity instead

File: app/src/main/java/com/google/droidjump/DrawableElement.java (hunk 6 above)
Comment (redls) [reply app/src/main/java/com/google/droidjump/DrawableElement.java]: I would say no, we don't

File: app/src/main/java/com/google/droidjump/Droid.java (hunk 9 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/Droid.java:40]: synced offline and agreed to resolve comment

File: app/src/main/java/com/google/droidjump/GameFragment.java [hunk 12]
```diff
@@ -33,9 +39,19 @@
...
-                screen);
-        gameView = new GameView(getActivity(), screen.x, screen.y, /* isPlaying= */ true);
+        Activity activity = getActivity();
+        WindowManager windowManager = Objects.requireNonNull(activity).getWindowManager();
+        Display defaultDisplay = windowManager.getDefaultDisplay();
+
+        // Writing size to the screen variable
+        defaultDisplay.getSize(screen);
+
+        // Extracting screen coordinates
```
Comment (saimonsaret) [inline app/src/main/java/com/google/droidjump/GameFragment.java:49]: nit: don't really see a point in extracting coordinates like that. You use them one time and "screen.x" is descriptive enough to use directly.

File: app/src/main/java/com/google/droidjump/GameView.java [hunk 13]
```diff
@@ -165,6 +134,40 @@ private void drawDroid(Canvas canvas) {
         canvas.drawBitmap(droid.getBitmap(), droid.getX(), droid.getY(), /* paint= */ null);
     }
 
+    private void updateDroidCoordinates() {
+        if (droid.isJumping()) {
+            if (droid.getY() + droid.getHeight() < droid.getInitialY() - droid.getJumpHeight()) {
+                droid.setJumping(false);
+            } else {
+                droid.useJumpingBitmap();
+
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameView.java:143]: nit: remove empty line.
